galley/pkg/config/analysis/analyzers/authz/authorizationpolicies.go

```diff
@@ -142,11 +143,32 @@ func hasMatchingPodsRunningIn(selector k8s_labels.Selector, setList []k8s_labels
 func (a *AuthorizationPoliciesAnalyzer) analyzeNamespaceNotFound(r *resource.Instance, c analysis.Context) {
 	ap := r.Message.(*v1beta1.AuthorizationPolicy)
 
-	for _, rule := range ap.Rules {
-		for _, from := range rule.From {
-			for _, ns := range append(from.Source.Namespaces, from.Source.NotNamespaces...) {
+	for i, rule := range ap.Rules {
+		for j, from := range rule.From {
+			for k, ns := range from.Source.Namespaces {
 				if !matchNamespace(ns, c) {
-					c.Report(collections.IstioSecurityV1Beta1Authorizationpolicies.Name(), msg.NewReferencedResourceNotFound(r, "namespace", ns))
+					m := msg.NewReferencedResourceNotFound(r, "namespace", ns)
+
+					pathKeyForLine := fmt.Sprintf(util.AuthorizationPolicyNameSpace, i, j, k)
+					if line, ok := util.ErrorLine(r, pathKeyForLine); ok {
+						m.Line = line
+					}
+
+					c.Report(collections.IstioSecurityV1Beta1Authorizationpolicies.Name(), m)
+				}
+			}
+
+			// Check source.notNamespaces
+			for k, ns := range from.Source.NotNamespaces {
+				if !matchNamespace(ns, c) {
+					m := msg.NewReferencedResourceNotFound(r, "namespace", ns)
+
+					pathKeyForLine := fmt.Sprintf(util.AuthorizationPolicyNameSpace, i, j, k)
+					if line, ok := util.ErrorLine(r, pathKeyForLine); ok {
+						m.Line = line
+					}
+
+					c.Report(collections.IstioSecurityV1Beta1Authorizationpolicies.Name(), m)
```

Comment:
The new loop is used to provide the index number of the notNamespaces array for the path key. It can also be done within one loop, however there should be a check condition to the index, like:
`if index >= len(from.Source.Namespaces):  nIndex = index - len(from.Source.Namespaces)`
because namespaces and notNamespaces are two different arrays in the YAML file.